### ISSUE TITLE
3DES/SWEET32

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8781]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)| @hynek reported|
|-|-|
|Trac ID|trac#8781|
|Type|defect|
|Created|2016-08-24 13:49:10Z|
|Branch|https://github.com/twisted/twisted/tree/8781-lukasa-byebye3des|

3DES should be considered broken and removed: https://www.openssl.org/blog/blog/2016/08/24/sweet32/

Main victim is IE8 on Win XP from 2011 here.

<details><summary>Searchable metadata</summary>

```
trac-id__8781 8781
type__defect defect
reporter__hynek hynek
priority__normal normal
milestone__None None
branch__8781_lukasa_byebye3des 8781-lukasa-byebye3des
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1472046550744550 1472046550744550
changetime__1472058937638197 1472058937638197
version__None None
owner__Cory_Benfield__lukasaoz_____ Cory Benfield <lukasaoz@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_2"></a>|@Lukasa set owner to @Lukasa|
|-|-|

I'll take this.

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_3"></a>|@Lukasa removed owner|
|-|-|

See https://github.com/twisted/twisted/pull/522

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_4"></a>|@markrwilliams commented|
|-|-|

While I think it's hilarious that Twisted was offering both ChaCha20 and 3DES as acceptable ciphers, I think it's time to retire 3DES.

Please merge once the builders are green!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban commented|
|-|-|

I have not yet fully read this, but I think that the ticket and relase notes (topfile) should make it clear that the change is only for SSL/TLS

conch.ssh still has 3des support ... not sure if we should remove 3des from ssh

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_6"></a>|@Lukasa commented|
|-|-|

Adi, the release note is as clear on that as I can imagine: "Client and server TLS connections made via the client TLS endpoint and the server SSL endpoint, as well as any other code that uses twisted.internet.ssl.CertificateOptions, no longer accept 3DES-based cipher suites by default, to defend against SWEET32.".

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_7"></a>|@adiroiban commented|
|-|-|

True. Sorry. My bad!

I have read the ticket description and the PR title and I was left with the impression that this the target of this change is not clear.

All good.

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_8"></a>|@Lukasa set owner to @Lukasa<br>@Lukasa set status to `closed`|
|-|-|

In changeset a1b802e6326e1fc67527fff0263544bb3d19312b
```
#!CommitTicketReference repository="" revision="a1b802e6326e1fc67527fff0263544bb3d19312b"
Merge 8781-lukasa-byebye3des: Remove 3DES support

Author: Lukasa
Reviewer: markrwilliams
Fixes: #8781
```
